Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -126,18 +130,33 @@ void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DO
     }
 
     auto& workerGlobalScope = downcast<WorkerGlobalScope>(*context);
-    auto completionHandler = [originData = WTFMove(originData).isolatedCopy(), permissionDescriptor, contextIdentifier = workerGlobalScope.identifier(), promise = WTFMove(promise)] (auto& context) mutable {
+    PermissionQuerySource source;
+
+    if (is<DedicatedWorkerGlobalScope>(workerGlobalScope))
+        source = PermissionQuerySource::DedicatedWorker;
+    else
+        if (is<SharedWorkerGlobalScope>(workerGlobalScope))
+            source = PermissionQuerySource::SharedWorker;
+        else
+            if (is<ServiceWorkerGlobalScope>(workerGlobalScope))
+                source = PermissionQuerySource::ServiceWorker;
+            else {
+                promise.reject(Exception { InvalidStateError, "The context is invalid"_s });
+                return;
+            }
+
+    auto completionHandler = [originData = WTFMove(originData).isolatedCopy(), permissionDescriptor, contextIdentifier = workerGlobalScope.identifier(), source, promise = WTFMove(promise)] (auto& context) mutable {
         ASSERT(isMainThread());
 
         auto& document = downcast<Document>(context);
-        if (!document.page()) {
+        if (source == PermissionQuerySource::DedicatedWorker && !document.page()) {
```

Comment:
I would have thought service workers/shared workers would still have a Page (just not a WebPage/WebPageProxy)

Reply:
You're right--Fixed!